modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicEventHandlers.java

```diff
@@ -240,6 +267,36 @@ protected void fireFileEvent(
         }
     }
 
+    /**
+     * Firing an event to listeners in order to inform them about what we are doing and about the
+     * percentage of work already carried out.
+     *
+     * @param inMessage The message to show.
+     * @param percentage The percentage for the process.
+     */
+    protected void fireUrlEvent(
+            Level level,
+            final URL url,
+            final boolean ingested,
+            final String inMessage,
+            final double percentage) {
+        if (LOGGER.isLoggable(level)) {
+            LOGGER.log(level, inMessage);
+        }
+        synchronized (notificationListeners) {
+            final String newLine = System.getProperty("line.separator");
+            final StringBuilder message = new StringBuilder("Thread Name ");
+            message.append(Thread.currentThread().getName()).append(newLine);
+            message.append(this.getClass().toString()).append(newLine).append(inMessage);
+            final URLProcessingEvent evt =
```

Comment:
Is this the only line different from fireFileEvent? If so, can you please factor out the difference and share code?